pkg/storegateway/storepb/indexheader.proto

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+syntax = "proto3";
+
+package thanos;
+
+option go_package = "storepb";
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+
+option (gogoproto.sizer_all) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+option (gogoproto.goproto_getters_all) = false;
+
+// Do not generate XXX fields to reduce memory footprint.
+option (gogoproto.goproto_unkeyed_all) = false;
+option (gogoproto.goproto_unrecognized_all) = false;
+option (gogoproto.goproto_sizecache_all) = false;
+
+message IndexHeaderTrackerState {
+  map<string, uint32> lazy_loaded_blocks = 1;
```

Comment:
What's the reason for using uint32? Maybe this code will still be in use [15 years from now](https://en.wikipedia.org/wiki/Year_2038_problem#:~:text=Consequently%2C%20if%20a%20signed%2032,on%20Tuesday%2C%2019%20January%202038.) 🤷 I think we should go with uint64 or an int64. Maybe a int64 will make using the existing `usedAt` atomic easier

Reply:
In discussion in design doc, somebody suggested to just store timestamp in second precision. Hence uint32 should enough.

Reply:
My point is that we shouldn't be using uint32 for unix timestamps because of the Y2038 problem. As for seconds/milliseconds/nanoseconds - I don't have a strong preference.

Reply:
Got it. My on top of the head calculation seems overestimate the range of uint32 for that. Good to know about Y2038.